Sources/Meta/Visitors/CopyVisitor.swift

```diff
@@ -0,0 +1,36 @@
+import SwiftSyntax
+
+public class CopyVisitor: SyntaxVisitor {
+    public private(set) var generatedCode = ""
+    
+    override public func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+        let structName = node.identifier.description.trimmingCharacters(in: .whitespacesAndNewlines)
+        let fields = node.fields
+        
+        let generated = """
+                        extension \(structName) {
+                            func copy(\(copyParameters(for: fields))) -> \(structName) {
+                                \(body(for: structName, with: fields))
+                            }
+                        }
+                        
+                        """
+        print(generated, to: &generatedCode)
```

Comment:
why do not directly assign in the private var `generatedCode`?

Reply:
There may be more structs in the field, so if we assign the value of `generatedCode`, we may overwrite previous computations. Usual problem with **mutability**.

Reply:
Ah! I did not know it does an append to the variable, maybe it will be better if we change by `generatedCode += ""` or `generatedCode = "\(generatedCode)\n\n\(newGeneratedCode)"` I am only worried about the intention to do not override previous value.